### ISSUE TITLE
Linux podman info - stat /bin/fuse-overlayfs: no such file or directory

### ISSUE THREAD

**Issue by Constantin07** [containers/podman#12613]
/kind bug

**Description**

On CentOS7.9 with `podman` build from sources it fails to find `fuse-overlayfs` under `/usr/local/bin`. This happens when invoked as non-privileged user.
As `root` user it works just fine.

Note: The `fuse-overlayfs` has been compiled from sources and installed in `/usr/local/bin`.
The `PATH` variable includes `/usr/local/bin` for both `root` and non-privileged user.

**Steps to reproduce the issue:**

1.

2.

3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
(paste your output here)
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by giuseppe** [comment]
does the user have a custom configuration file under `~/.config/containers` that overrides the mount_program?

**Comment by Constantin07** [comment]
You are right, @giuseppe I did found in `~/.config/containers` a storage config file though not sure how it was created. Probably from old podman rpm package which I did install initially and then removed because it was very old.

The above file indeed had `mount_program = "/bin/fuse-overlayfs"`. Now If I rename this file I get:
```bash
$ podman info
Error: error acquiring runtime init lock: error opening "/run/libpod/alive.lck": permission denied
```
Wasn't supposed podman to look into `/etc/containers/storage.conf` if user's config file is absent ?
root user doesn't have `~/.config/containers` and is able to use the "global" config file.

**Comment by rhatdan** [comment]
Looks like XDG_RUNTIME_DIR is set incorrectly.

**Comment by giuseppe** [comment]
if you don't override the configuration in `~/.config/containers` then what is set in `/etc/containers` is used.

Older versions of Podman were always creating files under `~/.config`.  Can you try removing `storage.conf` as well?

**Comment by Constantin07** [comment]
@rhatdan looks like that's the case. As `root` logged in on a box I get:
```bash
# echo $XDG_RUNTIME_DIR
/run/user/0
```

Now, when I do `sudu su - jenkins` I get that it's not set:
```bash
$ echo $XDG_RUNTIME_DIR

```
but I do get `XDG_SESSION_ID=2` env. var set. This is a login/interactive shell.
Now I'm wondering if `XDG_RUNTIME_DIR` is supposed to be set when run as service account, e.g. running Jenkins with `jenkins` non-priviledged user ?

**Comment by Constantin07** [comment]
@giuseppe I did remove entirely the `~/.config/containers` directory for regular user (let's call it `jenkins`).
but unfortunately no luck - getting the same error:
```bash
$ podman info
Error: error acquiring runtime init lock: error opening "/run/libpod/alive.lck": permission denied
```

Here is my global config:
```bash
# ls -la /etc/containers
-rw-r--r--  1 root root   74 Dec  9 20:24 policy.json
-rw-r--r--  1 root root  132 Dec 16 13:08 registries.conf
-rw-r--r--  1 root root  308 Dec 15 00:32 storage.conf

# cat policy.json
{
  "default": [
    {
      "type": "insecureAcceptAnything"
    }
  ]
}

# cat registries.conf
unqualified-search-registries = ['docker.io', 'centos7.internal:5000']
```

The `storage.conf` is as per dump shown above.

**Comment by Constantin07** [comment]
defining XDG_RUNTIME_DIR env. var. doesn't help:
```bash
$ export XDG_RUNTIME_DIR=/run/user/$UID
$ echo $XDG_RUNTIME_DIR
/run/user/995
$ podman -v
podman version 4.0.0-dev
$ podman info
Error: error acquiring runtime init lock: error opening "/run/libpod/alive.lck": permission denied
```

**Comment by rhatdan** [comment]
@mheon WDYT?

**Comment by mheon** [comment]
Please remove `~/.config/containers` again after ensuring that $XDG_RUNTIME_DIR is defined; Podman is using the old paths, cached in the DB, which contain an invalid tmp directory.

**Comment by Constantin07** [comment]
@mheon I did remove it. But I have the same result.
I've observed that it does create an empty file `~/.config/cni/net.d/cni.lock`.

Please note that I'm using `podman` only for building container images, not for running them.

The below error message is misleading: the file `/run/libpod/alive.lck` does exist, is empty and has read permission allowed for everyone. If it tries to open this file in write mode, probably it should be more explicit about this.

```
$ podman info
Error: error acquiring runtime init lock: error opening "/run/libpod/alive.lck": permission denied
-bash-4.2$ ls -la /run/libpod/alive.lck
-rw-r--r-- 1 root root 0 Dec 17 14:41 /run/libpod/alive.lck
```

**Comment by mheon** [comment]
That directory should not be in use by rootless Podman. We should be using a directory in `/run/user/$UID` (or, failing that, a directory in `/tmp` unique to the user in question). That is the root Podman temporary files directory. It cannot be used by rootless Podman.

**Comment by Constantin07** [comment]
Thanks @mheon for explaining, I'm wondering what other thing could make `podman` when run as privileged user to connect to too `root` resources ?

**Comment by rhatdan** [comment]
Since this is a setup issue on a non supported platform, moving this to discussion.
